File: server/DrawingApp/DrawingApp.DM/Drawing.cs
namespace DrawingApp.DM;

public class Drawing
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string CommandsJson { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public User? User { get; set; }
}

File: client/src/types/drawing.ts
export type Message = {
  text: string;
  type: 'user' | 'bot';
};

export type Shape = {
  Type: 'rect' | 'circle';
  X: number;
  Y: number;
  Color: string;
  Width?: number;
  Height?: number;
  Radius?: number;
};

export interface Drawing {
  id: number;
  name: string;
  commandsJson: string;
  createdAt: string;
  userId: number;
}

export interface CanvasSectionProps {
  canvasRef: React.RefObject<HTMLCanvasElement>;
  currentShapes: Shape[]; // Data passed down from App.tsx
}